src/mm/allocator.rs

```diff
@@ -97,12 +104,11 @@ impl Heap {
 		if self.bottom == 0 {
 			unsafe { self.alloc_bootstrap(layout) }
 		} else {
-			let mut size = layout.size();
-			if size < HoleList::min_size() {
-				size = HoleList::min_size();
-			}
-			let size = align_up!(size, mem::align_of::<Hole>());
-			let layout = Layout::from_size_align(size, layout.align()).unwrap();
+			let mut size = cmp::max(layout.size(), HoleList::min_size());
+			size = align_up!(size, mem::align_of::<Hole>());
+			size = align_up!(size, CACHE_LINE_SIZE);
+			let layout =
+				Layout::from_size_align(size, cmp::max(layout.align(), CACHE_LINE_SIZE)).unwrap();
```

Comment:
Will this force a minimum alignmend of CACH_LINE_SIZE for every allocation? Do you know how this is done on Linux or any other OS?

Reply:
Hm, normally the try to find smallest power-of-two block size big enough to hold requested size plus header.

https://github.com/lattera/glibc/blob/895ef79e04a953cac1493863bcae29ad85657ee1/hurd/hurdmalloc.c#L213-L223

Reply:
Thanks for the link. That got me thinking. I am not sure why they do it they way they do it and I think jemalloc does it differently. Then I was thinking if we should try to do something that realloc can be done efficiently. After that I was worried that the implementation will get to complex and we'll have more errors than before. I guess, I am just not a fan of having to implement your own allocator, even though I understand why it is necessary at the moment.

Long story short: Feel free to ignore this comment and if you ever have good idea on how to get rid of implementing our own allocator I am all yours 😉